pkg/scm/git/git.go

```diff
@@ -492,3 +556,27 @@ func pipeToLog(reader io.Reader, log func(...interface{})) {
 		}
 	}
 }
+
+type MissingGitBinaryError struct {
+}
+
+func (err MissingGitBinaryError) Error() string {
+	return "Git was not found in the PATH"
+}
+
+type EmptyGitRepositoryError struct {
+	source string // The source destination
+}
+
+func (err EmptyGitRepositoryError) Error() string {
+	return fmt.Sprintf("The git repository \"%s\" has no tracking information or commits", err.source)
+}
+
+type InvalidReferenceError struct {
+	ref    string // The invalid reference
+	source string // The input Git repo, with the ref
+}
+
+func (err InvalidReferenceError) Error() string {
+	return fmt.Sprintf("Invalid reference \"%s\" in %s", err.ref, err.source)
+}
```

Comment:
I'm ok with either!  Is that where all errors go?  It would make sense either way